### PR TITLE
Benchmark with app harness

### PR DESCRIPTION
**Pull request by ElijahAhianyo**
## Summary
- Add pytest benchmark on top of app harness test to generate stats on small/medium sized apps in CI.
- CI is run on linux/Mac/Windows for supported python major+minor versions and pushed to new table on RDB.

## Tests
- [x] new benchmark tests pushed results to RDB
image

### CODE REVIEW

File: benchmarks/test_benchmark_compile_components.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    import reflex as rx
+
+    def index() -> rx.Component:
+        return rx.center(rx.vstack(*render_component(1)))
+
+    app = rx.App(state=rx.State)
+    app.add_page(index)
+
+
+def AppWithHUndredComponentOnePage():
```
Comment (masenf) [inline benchmarks/test_benchmark_compile_components.py:126]:
```suggestion
def AppWithHundredComponentOnePage():
```

File: benchmarks/test_benchmark_compile_components.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+    app = rx.App(state=rx.State)
+    app.add_page(index)
+
+
+def AppWithThousandComponentsOnePage():
+    """A reflex app with 1000 components on a page."""
+    import reflex as rx
+
+    def index() -> rx.Component:
+        return rx.center(rx.vstack(*render_component(10)))
```
Comment (masenf) [inline benchmarks/test_benchmark_compile_components.py:142]:
```suggestion
        return rx.center(rx.vstack(*render_component(1000)))
```

1000 right?

**Review by masenf — CHANGES_REQUESTED** [review]
Replace "Test that background tasks work as expected." with the actual description.

**Comment by ElijahAhianyo** [comment]
> Replace "Test that background tasks work as expected." with the actual description.

Fixed 😅

**Review by masenf — APPROVED** [review]

**Pull request merged**